src-tauri/src/main.rs

```diff
@@ -43,28 +43,19 @@ fn plot_progress_tracker() -> usize {
 }
 
 #[tauri::command]
-async fn farming(path: String, reward_address: String, plot_size: u64) {
-    match reward_address.len() {
-        0 => {
-            farm(path.into(), "ws://127.0.0.1:9944", None, plot_size)
-                .await
-                .unwrap();
-        }
-        _ => {
-            if let Ok(address) = parse_reward_address(&reward_address) {
-                farm(path.into(), "ws://127.0.0.1:9944", Some(address), plot_size)
-                    .await
-                    .unwrap();
-            } else {
-                error!("Reward address could not be parsed!");
-            }
-        }
+async fn farming(path: String, reward_address: String, plot_size: u64) -> [u8; 32] {
```

Comment:
Why does it still return something?

Reply:
at the time, multi-replica is not merged to main. So, we need the public address from the backend to be able distinguish which blocks are farmed by us.

Reply:
So, this will not return anything in the multi-replica PR :)

Reply:
How is multi-replica related to this if reward address is supported since initial March release?

Reply:
For reward address, we need to return the farmer's public address to the frontend (to see which blocks are farmed by the farmer). 

Here is how multi-replica steps in: since there can be multiple plots, this function will return multiple public keys (and the received block will be checked against all of them).

My previous comment was wrong (i was confused), it will return `Vec<[u8; 32]>` in multi-replica PR

Reply:
I don't get why it needs to return anything either in multi or single replica. Just send reward address from frontend and you will always have the same reward address in produced blocks. Frontend doesn't need to know anything about backend identities.

Reply:
Hmm maybe its a deeper issue. 
Leo designed the frontend as such: it is listening to the new produced blocks. Then compares the author of these new produced block with the farmer's public key. 

Instead of comparing the authors, we can compare the reward address embedded in the blocks, is this what you are suggesting?

Reply:
Yes.

Thinking more about it, the only challenge with that is if there is another farmer producing blocks for the same reward address, then it will show blocks produced by another farmer, which may or may not be desirable.

Reply:
I also thought about that but the application is for displaying the rewards in the end. So I think it's not a problem. Will do the changes in the frontend for that